zedUpload/httputil/http.go

```diff
@@ -196,18 +203,44 @@ func ExecCmd(ctx context.Context, cmd, host, remoteFile, localFile string, objSi
 
 // execCmdGet executes the get command.
 // NOTE: These **must** be named return values, or our defer to modify them will not work.
-func execCmdGet(ctx context.Context, objSize int64, localFile string, host string, client *http.Client, prgNotify types.StatsNotifChan, inactivityTimeout time.Duration) (stats types.UpdateStats, rsp Resp) {
-	var copiedSize int64
+func execCmdGet(
+	ctx context.Context,
+	objSize int64,
+	localFile, host string,
+	client *http.Client,
+	doneParts types.DownloadedParts,
+	prgNotify types.StatsNotifChan,
+	inactivityTimeout time.Duration,
+) (stats types.UpdateStats, rsp Resp) {
+	copiedSize := doneParts.PartSize
 
 	stats.Size = objSize
 	dirErr := os.MkdirAll(filepath.Dir(localFile), 0755)
 	if dirErr != nil {
 		stats.Error = dirErr
 		return stats, Resp{}
 	}
-	local, fileErr := os.Create(localFile)
-	if fileErr != nil {
-		stats.Error = fileErr
+
+	var local *os.File
+	if _, err := os.Stat(localFile); err == nil {
+		local, stats.Error = os.OpenFile(localFile, os.O_RDWR, 0644)
+		if stats.Error != nil {
+			return stats, Resp{}
+		}
+		if _, err = local.Seek(copiedSize, io.SeekStart); err != nil {
+			local.Close()
+			stats.Error = err
+			return stats, Resp{}
+		}
+	} else if os.IsNotExist(err) {
+		f, ferr := os.Create(localFile)
+		if ferr != nil {
+			stats.Error = ferr
+			return stats, Resp{}
+		}
+		local = f
+	} else {
+		stats.Error = err
 		return stats, Resp{}
 	}
```

Comment:
Instead of first checking if the file exists and depending on that creating the file or open it, can't you just open it with `os.O_CREATE` (but without `os.O_TRUNC`)?

```suggestion
	local, stats.Error = os.OpenFile(localFile, os.O_RDWR | os.O_CREATE, 0644)
	if stats.Error != nil {
		return stats, Resp{}
	}

	if _, err = local.Seek(copiedSize, io.SeekStart); err != nil {
		local.Close()
		stats.Error = err
		return stats, Resp{}
	}
```